### PR TITLE
Make home var settable

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

**Comment by 6543** [comment]
using `docker buildx build --build-arg HOME=/other ...` does not respect the build argument :/

I have no idea why it's not respected

**Comment by 6543** [comment]
the ARG needs to be [after the FROM](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact) like in the [test-env image](https://gitea.com/gitea/test-env/src/branch/main/Dockerfile#L1) for instance.

File: docs.md [hunk 1]
```diff
@@ -50,5 +50,6 @@ clone:
 | `backoff`                 | `5sec`            | Change backoff duration
 | `attempts`                | `5`               | Change backoff attempts
 | `branch`                  | $CI_COMMIT_BRANCH | Change branch name to checkout to
+| `home`                    |                   | Change HOME var for commands executed, fail if not exist
```
Comment (lafriks) [inline docs.md:53]:
```suggestion
| `home`                    |                   | Change HOME var for commands executed, fail if it does not exist
```

File: main.go [hunk 2]
```diff
@@ -126,6 +124,11 @@ func main() {
 			Usage:   "Change branch name",
 			EnvVars: []string{"PLUGIN_BRANCH", "CI_COMMIT_BRANCH", "CI_REPO_DEFAULT_BRANCH"},
 		},
+		&cli.StringFlag{
+			Name:    "home",
+			Usage:   "Change home dir",
```
Comment (lafriks) [inline main.go:129]:
```suggestion
			Usage:   "Change home directory",
```

File: plugin.go [hunk 3]
```diff
@@ -315,3 +322,22 @@ func remapSubmodule(name, url string) *exec.Cmd {
...
+		home = xdg.Home
+	}
+
+	// make sure home dir exist and is set
+	homeExist, err := pathExists(home)
+	if err != nil {
+		return err
+	}
+	if !homeExist {
+		return fmt.Errorf("home dir '%s' do not exist", home)
```
Comment (lafriks) [inline plugin.go:338]:
```suggestion
		return fmt.Errorf("home directory '%s' do not exist", home)
```

**Review by lafriks — APPROVED** [review]

**Pull request merged**